libs/sql-parser/src/main/java/io/crate/sql/tree/CreateRole.java

```diff
@@ -21,24 +21,40 @@
 
 package io.crate.sql.tree;
 
-import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
-public class CreateRole<T> extends Statement {
+public class CreateRole extends Statement {
     private final String name;
-    private final List<String> options;
+    private final boolean isUser;
 
+    private final GenericProperties<Expression> properties;
+    private final Map<String, String> options;
 
-    public CreateRole(String name, List<String> options) {
+
+    public CreateRole(String name,
+                      boolean isUser,
+                      GenericProperties<Expression> properties,
+                      Map<String, String> options) {
```

Comment:
What's the difference between options and properties? Could use some docs

Reply:
True, the `options` is Postgres syntax, the `properties` is the syntax we had (common with CREATE TABLE, etc.) I'll add comments, thx.

Reply:
Couldn't we normalize this to have just one?

I also think it would be nice if we supported parameter placeholders in addition to string literals for the values. It's much simpler to use from scripts if a password can be a `?` in addition to a literal string.

Reply:
Thx, it makes sense, I'm already trying to unify to have a `Map<String, Expression>` for both syntaxes.